lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -248,58 +255,44 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
     if (!remoteOpPath)
       return failure();
     SmallVector<Attribute> refSendPath;
-    // Verbatim XMR begins with the Top level module.
-    refSendPath.push_back(refSendPathList[remoteOpPath.value()]
-                              .first.cast<InnerRefAttr>()
-                              .getModuleRef());
     SmallString<128> xmrString;
-    Attribute lastInnerRef;
     size_t lastIndex;
     unsigned index = 0;
     for (; remoteOpPath; ++index) {
       lastIndex = remoteOpPath.value();
       auto entr = refSendPathList[remoteOpPath.value()];
       refSendPath.push_back(entr.first);
-      lastInnerRef = entr.first;
       remoteOpPath = entr.second;
-      ("{{" + Twine(index) + "}}").toVector(xmrString);
-      xmrString += '.';
     }
-    ("{{" + Twine(index) + "}}").toVector(xmrString);
     auto iter = xmrPathSuffix.find(lastIndex);
+
     // If this xmr has a suffix string (internal path into a module, that is not
     // yet generated).
     if (iter != xmrPathSuffix.end())
       xmrString += ("." + iter->getSecond()).str();
-    if (auto vec = resolve.getResult().getType().dyn_cast<FVectorType>()) {
```

Comment:
Neat!

Reply:
I was happy that this whole thing disappears. (I was also somewhat nervous looking at this that this was special-casing a subindex user as meaning "memory".) It was really neat that this devolves to the base case and no modification of the subindex is necessary.